node/network/availability-distribution/src/requester/mod.rs

```diff
@@ -215,3 +253,69 @@ impl Stream for Requester {
 		}
 	}
 }
+
+/// Requests up to `limit` ancestor hashes of relay parent in the same session.
+async fn get_block_ancestors_in_same_session<Context>(
```

Comment:
Logic here seems accurate accounting for the fact that `get_session_index` is actually returning the session index of the child. Nice